src/MockCollection.php

```diff
@@ -80,24 +87,60 @@ public function updateOne($filter, $update, array $options = [])
         $matcher = $this->matcherFromQuery($filter);
         foreach ($this->documents as $i => &$doc) {
             if ($matcher($doc)) {
-                foreach ($update['$set'] ?? [] as $k => $v) {
-                    $doc[$k] = $v;
-                }
+                $this->updateCore($doc,$update);
                 return;
             }
         }
+        $this->updateUpsert($filter,$update,$options,false);
     }
 
     public function updateMany($filter, $update, array $options = [])
     {
         $matcher = $this->matcherFromQuery($filter);
+        $anyUpdates = false;
         foreach ($this->documents as $i => &$doc) {
             if (!$matcher($doc)) {
                 continue;
             }
 
-            foreach ($update['$set'] ?? [] as $k => $v) {
-                $doc[$k] = $v;
+            $this->updateCore($doc,$update); 
+            $anyUpdates = $anyUpdates || true;
```

Comment:
I don't get this. *"[something] or true"* should always be *true*. `$anyUpdates = true` should have the same effect.